mesonbuild/interpreterbase/helpers.py

```diff
@@ -54,3 +57,22 @@ def default_resolve_key(key: mparser.BaseNode) -> str:
     if not isinstance(key, mparser.IdNode):
         raise InterpreterException('Invalid kwargs format.')
     return key.value
+
+def stringifyUserArguments(args: TYPE_var, subproject: SubProject, quote: bool = False) -> str:
+    if isinstance(args, str):
+        return f"'{args}'" if quote else args
+    elif isinstance(args, bool):
+        return 'true' if args else 'false'
+    elif isinstance(args, int):
+        return str(args)
+    elif isinstance(args, list):
+        l = [stringifyUserArguments(x, subproject, True) for x in args]
+        return '[%s]' % ', '.join(l)
+    elif isinstance(args, dict):
+        l = [f'{stringifyUserArguments(k, subproject, True)} : {stringifyUserArguments(v, subproject, True)}' for k, v in args.items()]
```

Comment:
I find this kind of f-string where basically the entire string is just embedded python code, kind of difficult to read. :(

Reply:
done